client/extensions/woocommerce/app/settings/email/email-settings/index.js

```diff
@@ -12,28 +13,209 @@ import { connect } from 'react-redux';
+	},
+];
 
 class Settings extends React.Component {
 
+	fetchSettings = ( props ) => {
+		const { siteId, fetchSettings } = props;
+		siteId && fetchSettings( siteId );
+	}
+
 	componentDidMount = () => {
-		const { siteId, fetchEmailSettings: fetch } = this.props;
-		siteId && fetch( siteId );
+		this.fetchSettings( this.props );
 	};
 
-	componentWillReceiveProps = newProps => {
-		if ( newProps.siteId === this.props.siteId ) {
-			return;
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.siteId !== this.props.siteId ) {
+			this.fetchSettings( nextProps );
+		}
+
+		// Save settings request
+		if ( ( ! this.props.saveSettingsRequest ) && nextProps.saveSettingsRequest ) {
+			nextProps.loaded && nextProps.submit( nextProps.siteId, nextProps.settings );
+		}
+
+		// Settings save request finished.
+		if ( ( ! nextProps.isSaving ) && this.props.isSaving ) {
+			if ( nextProps.submitError ) {
+				nextProps.errorNotice( translate( 'There was a problem saving the email settings. Please try again.' ) );
+			} else {
+				nextProps.successNotice( translate( 'Email settings saved.' ), { duration: 4000 } );
+			}
 		}
+	}
+
+	onChange = ( event ) => {
+		const { onChange, siteId } = this.props;
+		onChange( siteId, event );
+	}
+
+	renderOriginNotification = ( item, index ) => {
+		const { settings, loading } = this.props;
+		return <NotificationsOrigin
+			key={ index }
+			item={ item }
+			isPlaceholder={ loading }
+			recipient={ get( settings, [ item.field, item.option, 'value' ], '' ) }
+			placeholder={ get( settings, [ item.field, item.option, 'default' ], '' ) }
+			onChange={ this.onChange }
+			checkEmail={ item.checkEmail }
+		/>;
+	}
 
-		const { siteId, fetchEmailSettings: fetch } = newProps;
-		siteId && fetch( siteId );
+	renderInternalNotification = ( item, index ) => {
+		const { settings, loading } = this.props;
+		return <InternalNotification
+			key={ index }
+			item={ item }
+			checked={ 'yes' === get( settings, [ item.field, 'enabled', 'value' ], '' ) }
+			recipient={ get( settings, [ item.field, 'recipient', 'value' ], '' ) }
+			placeholder={ get( settings, [ item.field, 'recipient', 'default' ], '' ) }
+			isPlaceholder={ loading }
+			onChange={ this.onChange }
+		/>;
+	}
+
+	renderCustomerNotification = ( item, index ) => {
+		const { settings, loading } = this.props;
+		return <CustomerNotification
+			key={ index }
+			item={ item }
+			isPlaceholder={ loading }
+			checked={ 'yes' === get( settings, [ item.field, 'enabled', 'value' ], '' ) }
+			onChange={ this.onChange }
+		/>;
 	}
 
 	/* eslint-disable wpcalypso/jsx-classname-namespace */
 	render() {
 		return (
 			<div className="email-settings__container">
+				<ExtendedHeader
+					label={ translate( 'Origin' ) }
+				/>
+				<List>
+					{ originNotifications.map( this.renderOriginNotification ) }
+				</List>
+				<div>
+					<ExtendedHeader
+						label={ translate( 'Internal notifications' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **50** times:
`translate( 'notifications' )` *ES Score: 9*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/email-settings-ui-save)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).